### DISCUSSION TITLE
Google Colab: "Is this normal?" and GPU selection

### DISCUSSION THREAD

**Discussion by erica-trump** [pyannote/pyannote-audio#1595] [Q&A]
**_Is this normal?_** On Colab, I am using a 16 GB RAM-GPU. I notice that the GPU usage _never_ goes above 2.7 GB RAM. It runs using 2.7 GB RAM for many hours. 

Some other info that would help:
- What is the best Colab GPU to use with pyannote?
- Any other "tips and tricks" for getting it to work?

Colab link: https://colab.research.google.com/drive/11tteAb_BkD8CIucQjogyOvgd7DiTazmd

**Background:**
I've been struggling with the **speaker-diarization-3.1** pipeline. I'm using this within a Google colab notebook. The job runs for 4+ hours and then times out. I think that something may be wrong with the setup of my environment. 

Prior to encountered the time-out issue, I struggled to get the right PyTorch configuration to work without dependency conflicts with pyannote. I had also previously used a "bad" configuration that made it so that the jobs wouldn't go to GPU at all. Here is what I have now:

torch Version: 2.1.0
torchvision Version: 0.16.2+cu118
tensorflow Version: 2.15.0.post1
protobuf Version: 4.23.4

**Comment by hbredin** [comment]
Without access to the actual Google Colab, solving your issue will be difficult.

**Reply by erica-trump** [reply]
Hi, good point! I'm very happy to share. I've been battling this for a long time at this point, and _really_ appreciate your help. 

here is the colab link: https://colab.research.google.com/drive/11tteAb_BkD8CIucQjogyOvgd7DiTazmd

The requirements file was pulled from github.

Here is its content:
asteroid-filterbanks >=0.4
einops >=0.6.0
huggingface_hub >= 0.13.0
lightning >= 2.0.1
omegaconf >=2.1,<3.0
pyannote.core >= 5.0.0
pyannote.database >= 5.0.1
pyannote.metrics >= 3.2
pyannote.pipeline >= 3.0.1
pytorch_metric_learning >= 2.1.0
rich >= 12.0.0
semver >= 3.0.0
soundfile >= 0.12.1
speechbrain >= 0.5.14
tensorboardX >= 2.6
torch >= 2.0.0
torch_audiomentations >= 0.11.0
torchaudio >= 2.0.0
torchmetrics >= 0.11.0

**Reply by hbredin** [reply]
Thanks. However, I am afraid this Google Colab does not allow me to reproduce your issue as it relies on files I don't have access to.

**Reply by erica-trump** [reply]
Hi,

Thank you for following up. I really appreciate your help in resolving this!

My Colab notebook uses two input files. 

1. A requirements.txt file. I pulled this file from the [pyannote-audio GitHub](https://github.com/pyannote/pyannote-audio) and shared the contents of in my last message.
2. A .wav file that was download from a YouTube video. I have used the pipeline on several .wav files that are needed for my project (long-form podcasts). I continually run into the same issue: the GPU RAM usage peaks at 2.7 GB (although 16 GB are available) and the diarization does not finish within a 4-hour timeframe. 

One thing I would really like to understand is whether you would expect more of the available GPU RAM to be used by the pipeline?

I'd also like to provide some some additional detail for the .wav file that should help you to reproduce the issue.

1. I am using yt_dlp to download the audio of long-form podcast YouTube videos. I also use pydub to add some silence at the start of the audio file.
2. The wav files are large (around 1.5 GB) with 2-3 hours of audio. 
4. I can share these files with you directly from my Google Drive, if that’s easiest or you. Just let me know how!
5. You can also produce them yourself using this Colab Notebook [linked here](https://colab.research.google.com/drive/1-1BQTANA2x5c1cdmvCXrDpxCpnPqy6u7?usp=sharing). At the top I have the URLs for three example YouTube videos. All result in the same behavior.

Thanks again. I truly appreciate the help.

**Reply by grr8coder** [reply]
Same issue on my side, on my local system it takes no time to get the answer but even for simple 3min sample on colab it takes forever.
